### ISSUE TITLE
no ICCID(Integrate circuit card identity) support?

### ISSUE THREAD

**Issue by yufengzjj** [arthurdejong/python-stdnum#58]

**Comment by arthurdejong** [comment]
This number is currently not supported. I've had a quick look and there are a number of issues with implementing this number:

- a lot of things are vague about the format (e.g. some places mention the number is 19 or 20 digits, some suggest that it may be shorter and others say it can be up to 22 digits)
- a lot of numbers that are printed on SIM cards either lack a check digit or have an extra digit appended (about half the numbers I found)
- the ICCID contains a IIN but the information available on https://www.itu.int/pub/T-SP-E.118-2015 is either outdated or incomplete (e.g. missing non-international operators and the US also has IINs that start with 8901 none of which are listed in the document)
- the complete IIN list is not public: https://en.wikipedia.org/wiki/ISO/IEC_7812#IIN_Register

This means there is not really a meaningful test that can be implemented that would validate a large number of ICCIDs.

[Here is a list of numbers I found](https://github.com/arthurdejong/python-stdnum/files/1596447/iccid.txt) and the various problems I found with them. If you have more information or some pointers on how to approach this to do meaningful validation it is welcome.

**Comment by AboTyim** [comment]
IMEI.info:
The format of the ICCID is: MMCC IINN NNNN NNNN NN Cx

    MM = Constant (ISO 7812 Major Industry Identifier)
    CC = Country Code
    II = Issuer Identifier
    N{12} = Account ID ("SIM number")
    C = Checksum calculated from the other 19 digits using the Luhn algorithm.
    x = An extra 20th digit is returned by the 'AT!ICCID?' command, but it is not officially part of the ICCID
    
source:
https://www.imei.info/news/iccid-checker/

**Comment by ValZapod** [comment]
> The format of

This is wrong. Country code can be 1 digit, for example for Russia it is just 7 (which is coming from our calling code +7, shared with Kasachstan, but Kasachstan uses 996 for Country code in ICCID.

All  ICCIDs are available from ITU (see SIM Card wikipedia article).

**Comment by ValZapod** [comment]
> * the complete IIN list is not public: https://en.wikipedia.org/wiki/ISO/IEC_7812#IIN_Register

This is not SIM cards. This is about credit and debit cards. Bank ones. First 6 digits of a credit card number is a BIN, Bank identification number.

For example, you can look bank cards up here https://binlist.net/ by first 6 digits.
